### DISCUSSION TITLE
Is it possible to specify Python version

### DISCUSSION THREAD

**Discussion by satwikkansal** [pyodide/pyodide#2353] [Q&A] [answered]
I'm wondering if there's a way to use a different Python version for the interpreter than the one bundled.

**Comment by rth** [comment, marked as answer]
No, currently Pyodide distribution is built for only 1 Python version:
 -  the upcoming 0.20 release will include Python 3.10
 -  Pyodide 0.18 - 0.19 had Python 3.9
 -  Pyodide 0.16.1 - 0.17 had Python 3.8

so currently the only way to switch Python version is to downgrade to an earlier release, but you will then lose access to all the new features and bug fixes.

It takes a lot of effort to support a given Python version, so it is unlikely that we will simultaneously support more than one in the near future.

**Reply by bertcoerver** [reply]
@rth do you maybe know if there is a table or something that specifies which python version is used for each pyodide release?

**Reply by rth** [reply]
@bertcoerver Not as far as I know (but I'm not following this project much anymore). Maybe better to open a new issue about it, if you need it.

**Reply by agriyakhetarpal** [reply]
> @rth do you maybe know if there is a table or something that specifies which python version is used for each pyodide release?

Hi @bertcoerver, we don't have a table per se, but you could take a look at this line in this file https://github.com/pyodide/pyodide/blob/6d26b3381f91152966744ba764d62c875d28b999/Makefile.envs#L1 across the Pyodide version tags to find out the Python version. In case you're building wheels for Pyodide using `pyodide-build`, then you may use `pyodide config list` to get a list of commonly used values, and you can get an individual value out of them as well, via `pyodide config get python_version`.

**Comment by satwikkansal** [comment]
Got it, thanks!
